File: Pharmacy_project/frontend_proj/src/services/medicineService.js
import api from './api';

export const getMedicines = () => api.get('/viewMedicine');
export const addMedicine = (data) => api.post('/addMedicine', data);
export const updateMedicine = (id, data) => api.put(`/updateMedicine/${id}`, data);
export const deleteMedicine = (id) => api.delete(`/deleteMedicine/${id}`);

File: Pharmacy_project/project_BE/project_crud/project_crud/src/main/java/com/project_crud/service/MedicineService.java
package com.project_crud.service;

import java.util.List;

import com.project_crud.model.Medicine;

public interface MedicineService {

    // Add Medicine
    public Medicine addMedicine(Medicine medicine);

    // View Medicines
    public List<Medicine> viewMedicines();

    // Delete Medicine
    public String deleteMedicine(int id);

    // Update Medicine
    public Medicine updateMedicine(int id, Medicine medicine);
}